### ISSUE TITLE
Poor Quality on Demo - Comparing Results on Paper

### ISSUE THREAD

**Issue by adrian-ash** [NiFangBaAGe/CoordFill#4]
Hello, 
Thank you for the great work. The results are impressive in terms of quality and performance at such a high resolution.

I ran the project on my local, both on CPU and CUDA. However, I can not reproduce the results in the paper with the given demo.py and pre-trained models.

There is some resolution/quality/blurriness problem in the output when I run the demo with the following command:

`python demo.py --config configs/test/test-Place2.yaml --model pretrained/train_Place2/encoder-epoch-last.pth --input example/3.png  --mask  example/3_mask.png --resolution 1024,1024`

The input image and mask are at 1024x1024 resolution. I am attaching the following files for your review:

The comparison image that shows from left to right (Original Image, Mask, Output Cordfill, Output Lama)
![overall_comparison_1](https://user-images.githubusercontent.com/132953935/236949627-08bc521a-ca5f-47d1-a801-9b7514b5114b.jpg)

Closeup to the problem. 
![grid_artifacts_poor_quality](https://user-images.githubusercontent.com/132953935/236949623-1ac882f4-121c-46d6-9390-d475c78476cb.jpg)

Original image and mask that is used. 
![3](https://user-images.githubusercontent.com/132953935/236949758-af92d0b4-7a96-4aed-b09e-75475ede4565.png)
![3_mask](https://user-images.githubusercontent.com/132953935/236949770-5445d272-8df2-4ae9-b458-90a9ef81cd37.png)

Please let me know if you encountered this kind of problem before. Maybe I am missing some input parameter? This issue happens on all the images that I tried. It feels like the low res features are not correctly upsampled by the pixel-wise query network and copied in low quality. But I could not find the exact problem.

I appreciate your help.

**Comment by NiFangBaAGe** [comment]
Thank you for your interest in our work.
CoordFill is limited in dealing with complex scenes. We downsample the input to 256x256 hence the original high-frequency details are hard to preserve. As shown in the above example, CoordFill has difficulty generating visual-pleasing pixels when there are many thin line-like objects in the background. They are tiny in the low-resolution image and can not be parameterized and queried accurately.
